public/js/presentation/controllers/viewableCtrl.ts

```diff
@@ -12,66 +29,193 @@ module PresentationApp.Controllers {
 
-		static $inject = ["$scope"];
-		constructor($scope: ng.IScope){
+		static $inject = ["$scope", "$timeout"];
+		constructor($scope: ng.IScope, $timeout: ng.ITimeoutService){
 			this.scope = $scope;
+			
+			// setTimeout doesn't digest angular scope changes, $timeout does
+			this.timeout = $timeout;
+			
+			this.slides = [];
+			this.isLoading = false;
 
+			// Other components communicate with this one through window.postMessage
+			// This setup cuts down on the number of requests to the server and ensures
+			// that the presentation stays in perfect sync with the presenter view.
 			window.addEventListener("message", (event) => {
 				if (event.origin !== Config.HOST) { return; }
 				var order = JSON.parse(event.data);
 				switch(order.action){
 
 					case "changeSlide":
-						this.slideUrl = order.data;
-						this.overlayActive = this.qaActive = false;
+						this.changeSlide(order.data);
 						break;
 
 					case "showOverlay":
-						this.overlayUrl = order.data;
-						this.qaActive = this.overlayIsVideo = false;
-						this.overlayActive = true;
+						this.showOverlay(order.data, false);
 						break;
 
 					case "showOverlayVideo":
-						this.overlayUrl = order.data;
-						this.qaActive = false;
-						this.overlayActive = this.overlayIsVideo = true;
+						this.showOverlay(order.data, true);
 						break;
 					
 					case "hideOverlay":
-						this.overlayUrl = undefined;
-						this.overlayActive = this.overlayIsVideo = false;
+						this.hideOverlay();
 						break;
 
 					case "showQASidebar":
-						this.question = JSON.parse(order.data);
-						this.overlayActive = false;
-						this.qaActive = true;
+						this.showQA(<FFQuestion>JSON.parse(order.data));
 						break;
 						
 					case "hideQASidebar":
-						this.question = undefined;
 						this.qaActive = false;
 						break;
 
 					case "showAccessLink":
 						this.accessLink = order.data;
+						
+						// Make sure the box is updated to footer size
+						setTimeout(function(){
+							window.dispatchEvent(new Event("updateFloatingContent"));
+						},10);
 						break;
 				}
 				$scope.$apply();
 			});
 		}
+		
+		// Proceed to a new slide
+		changeSlide(url: string, forwards: boolean = true) {
+			if (this.overlayActive) {
+				this.hideOverlay();
+				this.timeout(this.reallyChangeSlide.bind(this, url, forwards), 800);
+			} else if (this.qaActive) {
+				this.qaActive = false;
+				this.timeout(this.reallyChangeSlide.bind(this, url, forwards), 600);
+			} else {
+				this.reallyChangeSlide(url, forwards);
+			}
+		}
+			
+		private reallyChangeSlide(url: string, forwards: boolean = true) {
+			// First step is to load the image
+			var slideImage = new Image;
+			slideImage.addEventListener("load", ()=> {
+				this.scope.$apply(()=>{
+					this.isLoading = false;
+					
+					// Make a slide object and send it off to Angular
+					this.slides.push({
+						url: url,
+						width: slideImage.width,
+						height: slideImage.height
+					});
```

Comment:
Yes but Angular limits the rendering to 2 slides. This is because the transition animation requires that 2 slides briefly be visible at once. I wouldn't be surprised if that's the cause of the problem though, maybe once it gets those two it won't accept new ones, or something